src/test/kotlin/com/mparticle/kits/AppboyKitTest.kt

```diff
@@ -1290,4 +1290,75 @@ class AppboyKitTests {
         TestCase.assertEquals(0, currentUser.getCustomUserAttribute().size)
 
     }
+
+    @Test
+    fun testPurchase_Forward_product_name() {
+        var settings = HashMap<String, String?>()
+        settings[AppboyKit.APPBOY_KEY] = "key"
+        settings[AppboyKit.REPLACE_SKU_AS_PRODUCT_NAME] = "True"
+        val kit = MockAppboyKit()
+
+        kit.configuration =
+            KitConfiguration.createKitConfiguration(JSONObject().put("as", settings))
+        kit.onKitCreate(settings, MockContextApplication())
+        val product = Product.Builder("product name", "sku1", 4.5)
+            .build()
+        val commerceEvent = CommerceEvent.Builder(Product.CHECKOUT, product)
+            .currency("Moon Dollars")
+            .build()
+        kit.logTransaction(commerceEvent, product)
+        val braze = Braze
+        val purchases = braze.purchases
+        Assert.assertEquals(1, purchases.size.toLong())
+        val purchase = purchases[0]
+        Assert.assertEquals("product name", purchase.sku)
+        Assert.assertNull(purchase.purchaseProperties.properties[CommerceEventUtils.Constants.ATT_ACTION_CURRENCY_CODE])
+    }
+
+    @Test
+    fun testPurchase_Forward_product_name_When_flag_IS_FALSE() {
+        var settings = HashMap<String, String?>()
+        settings[AppboyKit.APPBOY_KEY] = "key"
+        settings[AppboyKit.REPLACE_SKU_AS_PRODUCT_NAME] = "False"
```

Comment:
can you add a test for when it's not in the settings at all?

Reply:
The test case has already been added